src/controllers/peers.ts

```diff
@@ -33,6 +34,13 @@ export default function (server: Hapi.Server, deps: Injector) {
     }
   }
 
+  async function validatePeer (request: Hapi.Request, h: Hapi.ResponseToolkit) {
+    const advertisedUrl = url.parse(request.server.info.uri)
+    if (request.server.info.host !== advertisedUrl.host) {
```

Comment:
What exactly is this validating? The only time these would not be equal is if you're connecting to a cname url which seems like a case we would want to support any way.

Reply:
So this would not add a peer who has forgotten to put CODIUS_PUBLIC_URI and also a host which is putting a CODIUS_PUBLIC_URI that is not their actual host.

Reply:
The idea is that the url would be https://codius.example.com so if a host discovered a peer from the default http://local.codius.org:PORT it would not match the host which is derived from the env since we are not setting it. https://github.com/hapijs/hapi/blob/master/API.md#server.options.host

Reply:
The goal is to not keep peers like local.codius.org in the discovered list, so I think in that case these would not be equal. However its still a good point we need to consider.